google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java

```diff
@@ -78,16 +180,42 @@ private StreamWriter(Builder builder) throws IOException {
       client = builder.client;
       ownsBigQueryWriteClient = false;
     }
-    connectionWorker =
-        new ConnectionWorker(
-            builder.streamName,
-            builder.writerSchema,
-            builder.maxInflightRequest,
-            builder.maxInflightBytes,
-            builder.limitExceededBehavior,
-            builder.traceId,
-            client,
-            ownsBigQueryWriteClient);
+    if (builder.connectionMode == ConnectionMode.SINGLE_TABLE) {
+      this.singleConnectionOrConnectionPool =
+          SingleConnectionOrConnectionPool.ofSingleConnection(
+              new ConnectionWorker(
+                  builder.streamName,
+                  builder.writerSchema,
+                  builder.maxInflightRequest,
+                  builder.maxInflightBytes,
+                  builder.limitExceededBehavior,
+                  builder.traceId,
+                  client,
+                  ownsBigQueryWriteClient));
+    } else {
+      // Assume the connection in the same pool share the same client and trace id.
+      // The first StreamWriter for a new stub will create the pool for the other
+      // streams in the same region, meaning the per StreamWriter settings are no
+      // longer working unless all streams share the same set of settings
+      this.singleConnectionOrConnectionPool =
+          SingleConnectionOrConnectionPool.ofConnectionPool(
+              connectionPoolMap.computeIfAbsent(
+                  ConnectionPoolKey.create(client),
+                  (key) ->
+                      new ConnectionWorkerPool(
+                          builder.maxInflightRequest,
+                          builder.maxInflightBytes,
+                          builder.limitExceededBehavior,
+                          builder.traceId,
+                          client,
+                          ownsBigQueryWriteClient)));
+      if (!Objects.equals(
```

Comment:
Let's also check limitExceedBehavior and ownsBigQueryWriteClient.

Reply:
Done, I think for ownsBigQueryWriteClient let's check the client instead